app/services/draft_edition_updater.rb

```diff
@@ -0,0 +1,19 @@
+class DraftEditionUpdater < EditionService
+
+  def perform!
+    if can_perform?
+      notify!
+      true
+    end
+  end
+
+  def failure_reason
```

Comment:
might be worth also adding a guard condition to fail if `state != 'draft'`? That would help to flag up coding errors early.